articles/libraries/auth0js/index.md

```diff
@@ -39,595 +39,182 @@ Install via [bower](http://bower.io):
+### Custom Username and Password
 
-If you're building a SPA (Single Page Application) and using Redirect Mode, then your `callbackURL` should send the user back to the same page. And because the `responseType` initialization option was set to `'token'`, Auth0 will also append a hash to that URL that will contain an `access_token` and `id_token` (the JWT). After control returns to your app, the full user profile can be retrieved via the `parseHash` and `getProfile` methods:
+The `client.login` method can be used when you need to request a username and password and authenticate from your site, instead of using the [hosted login page](/hosted-pages/login). This method can be used for database and ldap connections.
 
 ```js
-$(function () {
-  var result = auth0.parseHash(window.location.hash);
-
-  //use result.idToken to call your rest api
-
-  if (result && result.idToken) {
-    // optionally fetch user profile
-    auth0.getProfile(result.idToken, function (err, profile) {
-      alert('hello ' + profile.name);
-    });
-
-    // If offline_access was a requested scope
-    // You can grab the result.refresh_token here
-
-  } else if (result && result.error) {
-    alert('error: ' + result.error);
-  }
+webAuth.client.login({
+  realm: 'Username-Password-Authentication', //connection name or HRD domain
+  username: 'info@auth0.com',
+  password: 'areallystrongpassword',
+  audience: 'https://mystore.com/api/v2',
+  scope: 'read:order write:order',
+  }, function(err, authResult) {
+    // Auth tokens in the result or an error
 });
 ```
 
-If the `scope` option used with the `login` method did not contain `openid profile`, then the profile will only contain `user_id`.  In that case just parse the hash to obtain the user ID:
+### Custom Social Login
 
-```js
-$(function () {
-    var result = auth0.parseHash(window.location.hash);
-    if (result && result.profile) {
-      alert('your user_id is: ' + result.profile.sub);
-      //use result.id_token to call your rest api
-    }
-  });
-});
-```
-
-If there is no hash, `result` will be null.  If the hash contains the JWT, the `profile` field will be populated.
-
-##### Regular Web Apps
-
-If you're building a regular web application (HTML pages rendered on the server), then `callbackURL` should point to a server-side endpoint that will process the successful login, primarily to set some sort of session cookie. In this scenario you should make sure the `responseType` option is `'code'` (or just not specified) when the Auth0 client is created:
+- **client.login(options, cb)**: Authenticates the user with username & password in a realm using `/oauth/token`. This will not initialize a SSO session in auth0, hence can not be used along with renew auth.
 
 ```js
-var auth0 = new Auth0({
-  domain:       'mine.auth0.com',
-  clientID:     'dsa7d77dsa7d7',
-  callbackURL:  'http://my-app.com/callback'
-  // responseType not set (defaults to 'code')
+webAuth.client.login({
+  realm: 'db-conn'
```

Comment:
right, in that case should be like the example using authorize on the `Redirect Mode` section